File: app/src/main/java/ru/ra66it/updaterforspotify/presentation/ui/fragment/SettingsFragment.java
package ru.ra66it.updaterforspotify.presentation.ui.fragment;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.CheckBoxPreference;
import android.preference.Preference;
import android.preference.PreferenceFragment;
import android.preference.PreferenceManager;

import javax.inject.Inject;

import ru.ra66it.updaterforspotify.BuildConfig;
import ru.ra66it.updaterforspotify.UpdaterApp;
import ru.ra66it.updaterforspotify.R;
import ru.ra66it.updaterforspotify.data.storage.QueryPreferences;


/**
 * Created by 2Rabbit on 29.09.2017.
 */

public class SettingsFragment extends PreferenceFragment implements SharedPreferences.OnSharedPreferenceChangeListener {

    CheckBoxPreference prefEnableNotificationsOrigin;
    Preference prefAppInfo;
    SharedPreferences prefs;

    @Inject
    QueryPreferences queryPreferences;

    public static SettingsFragment newInstance() {
        return new SettingsFragment();
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        UpdaterApp.getApplicationComponent().inject(this);
        addPreferencesFromResource(R.xml.settings);

        prefs = PreferenceManager.getDefaultSharedPreferences(getActivity());

        prefEnableNotificationsOrigin = (CheckBoxPreference) findPreference("autoOrigin");
        prefEnableNotificationsOrigin.setChecked(queryPreferences.isEnableNotification());

        prefAppInfo = findPreference("verPref");
        prefAppInfo.setSummary(BuildConfig.VERSION_NAME);

    }

    @Override
    public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String s) {
        queryPreferences.setEnableNotification(
                prefEnableNotificationsOrigin.isChecked());

    }

    @Override
    public void onResume() {
        super.onResume();
        prefs.registerOnSharedPreferenceChangeListener(this);
    }

    @Override
    public void onStop() {
        super.onStop();
        prefs.unregisterOnSharedPreferenceChangeListener(this);
    }
}

File: app/src/main/java/ru/ra66it/updaterforspotify/presentation/ui/fragment/SettingsFragment.kt
package ru.ra66it.updaterforspotify.presentation.ui.fragment

import android.os.Bundle
import androidx.preference.CheckBoxPreference
import androidx.preference.ListPreference
import androidx.preference.Preference
import androidx.preference.PreferenceFragmentCompat
import ru.ra66it.updaterforspotify.R
import ru.ra66it.updaterforspotify.UpdaterApp
import ru.ra66it.updaterforspotify.presentation.viewmodel.SettingsViewModel
import javax.inject.Inject

class SettingsFragment : PreferenceFragmentCompat() {

    private var prefEnableNotificationsOrigin: CheckBoxPreference? = null
    private var prefCheckInterval: ListPreference? = null
    private var prefAppInfo: Preference? = null

    @Inject
    lateinit var viewModel: SettingsViewModel

    override fun onCreatePreferences(p0: Bundle?, p1: String?) {
        addPreferencesFromResource(R.xml.settings)
        UpdaterApp.applicationComponent.inject(this)

        prefEnableNotificationsOrigin = findPreference("updateCheckBox")
        prefEnableNotificationsOrigin?.let { enablePreference ->
            enablePreference.isChecked = viewModel.isEnableNotification
            enablePreference.setOnPreferenceChangeListener { _, newValue ->
                viewModel.toggleEnableNotifications(newValue)
                true
            }
        }

        prefCheckInterval = findPreference("updateInterval")
        prefCheckInterval?.let { checkInterval ->
            checkInterval.value = viewModel.checkIntervalDay
            checkInterval.setOnPreferenceChangeListener { _, newValue ->
                viewModel.updateInterval(newValue)
                true
            }
        }

        prefAppInfo = findPreference("verPref")
        prefAppInfo?.summary = viewModel.versionApp
    }

    companion object {
        fun newInstance(): SettingsFragment {
            return SettingsFragment()
        }
    }
}
